<launch>
    <!-- 启动全局规划算法 -->
    <node pkg="prometheus_local_planner" name="local_planner_main" type="local_planner_main" output="screen">
        <!-- 参数 -->
        <param name="uav_id" value="1"/>
        <param name="local_planner/sim_mode" value="true"/>
        <!-- 局部避障算法: 0为APF,1为VFH -->
        <param name="local_planner/algorithm_mode" value="0"/>
        <!-- 地图输入模式 0代表全局点云，1代表局部点云，2代表激光雷达scan数据 -->
        <param name="local_planner/map_input_source" value="0"/>
        <param name="local_planner/local_pcl_topic_name" value="/map_generator/local_cloud"/>
        <!-- 无人机飞行高度，建议与起飞高度一致 -->
        <param name="local_planner/fly_height" value="1.0"/>
        <!-- 最大速度 -->
        <param name="local_planner/max_planning_vel" value="0.5"/>
        <!-- APF参数 -->
		<!-- 膨胀参数，一般设置为无人机的半径或更大 -->
		<param name="apf/inflate_distance" value="0.4" type="double"/>
		<!-- 感知距离，只考虑感知距离内的障碍物 -->
		<param name="apf/obs_distance" value="5.0" type="double"/>  
		<!-- 增益 -->
		<param name="apf/k_repulsion" value="2.0" type="double"/>
		<param name="apf/k_attraction" value="1.0" type="double"/>
		<!-- 安全距离，距离障碍物在安全距离内，k_repulsion自动增大 -->
		<param name="apf/min_dist" value="0.1" type="double"/>
		<!-- 最大吸引距离 -->
		<param name="apf/max_att_dist" value="4" type="double"/>
		<!-- 地面高度，不考虑低于地面高度的障碍物 -->
		<param name="apf/ground_height" value="0.3" type="double"/>
		<!-- 地面安全高度，小于该高度，会产生向上推力 -->
		<param name="apf/ground_safe_height" value="0.3" type="double"/>
		<!-- 停止距离，小于该距离，停止自动飞行 -->
		<param name="apf/safe_distance" value="0.01" type="double"/>
    </node>
</launch>

